{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f256aff2-b4a3-4272-97a0-bd9e23c13a5b",
   "metadata": {},
   "source": [
    "# Scan for TF binding motifs and generate base GRN\n",
    "\n",
    "This notebook demonstrates how to scan for TF binding motifs. The base GRN will be generated by combining the ATAC-seq peaks and motif information.\n",
    "\n",
    "Much of this code is from CellOracle tutorial: https://morris-lab.github.io/CellOracle.documentation/notebooks/02_motif_scan/02_atac_peaks_to_TFinfo_with_celloracle_20200801.html\n",
    "\n",
    "### 0. Import libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "20a69dbf-c0c1-4374-b914-f4140ac0edae",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "\n",
    "import seaborn as sns\n",
    "\n",
    "import os, sys, shutil, importlib, glob\n",
    "from tqdm.notebook import tqdm\n",
    "\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "%matplotlib inline\n",
    "\n",
    "plt.rcParams['figure.figsize'] = (15,7)\n",
    "plt.rcParams[\"savefig.dpi\"] = 600"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "46a57586-52e3-4190-afa8-63281de3ee1d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.14.0'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "import celloracle as co\n",
    "from celloracle import motif_analysis as ma\n",
    "from celloracle.utility import save_as_pickled_object\n",
    "co.__version__"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f859176a-174a-457f-9bec-1845298f5cc4",
   "metadata": {},
   "source": [
    "### 1. Reference genome data preparation\n",
    "Before starting the analysis, we need to make sure the reference genome data is installed with genomepy. If not, please install the correct reference genome using the instructions below.\n",
    "\n",
    "genomepy installs reference genome data under home directory. But if you have installed or want to install reference genome to another specific location, please specify the place using genomes_dir argument."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fc2b1319-c6ef-4a9b-a856-3145cb5f63b2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mm10 installation:  True\n",
      "mm10 is installed.\n"
     ]
    }
   ],
   "source": [
    "ref_genome = \"mm10\"\n",
    "# genomes_directory = \"SET GENOMES DIRECTORY HERE\"\n",
    "genomes_directory = \"/project/clark-saucerman/genomes\"\n",
    "\n",
    "\n",
    "genome_installation = ma.is_genome_installed(ref_genome=ref_genome,\n",
    "                                             genomes_dir=genomes_directory)\n",
    "print(ref_genome, \"installation: \", genome_installation)\n",
    "\n",
    "if not genome_installation:\n",
    "    import genomepy\n",
    "    genomepy.install_genome(name=ref_genome, provider=\"UCSC\", genomes_dir=genomes_directory)\n",
    "else:\n",
    "    print(ref_genome, \"is installed.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e234522-eae8-4398-9c9c-7946cb438737",
   "metadata": {},
   "source": [
    "### 2. Load data\n",
    "In this notebook, we explain how to make a base GRN.\n",
    "\n",
    "Using the `processed_peak_file.csv` data file that was output from `6-tss-annotation.ipynb`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a4987b2a-1c83-44a3-b62c-773151a2afe9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>peak_id</th>\n",
       "      <th>gene_short_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100015519_100016019</td>\n",
       "      <td>Kitl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_100487279_100487779</td>\n",
       "      <td>Tmtc3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_100487943_100488443</td>\n",
       "      <td>Cep290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_100487943_100488443</td>\n",
       "      <td>Tmtc3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_100589013_100589513</td>\n",
       "      <td>4930430F08Rik</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     peak_id gene_short_name\n",
       "0  chr10_100015519_100016019            Kitl\n",
       "1  chr10_100487279_100487779           Tmtc3\n",
       "2  chr10_100487943_100488443          Cep290\n",
       "3  chr10_100487943_100488443           Tmtc3\n",
       "4  chr10_100589013_100589513   4930430F08Rik"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_dir = \"data/base_grn_outputs/E9/\"\n",
    "# wt_or_ko = \"KO\"\n",
    "peaks = pd.read_csv(f\"{data_dir}/processed_peak_file.csv\", index_col=0)\n",
    "peaks.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "850881ac-3383-4bbb-9755-7a9f92225585",
   "metadata": {},
   "source": [
    "Here, the function below will check peak data format, including chromosome name and peak length."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "add21ca9-9958-4e61-8d30-5e61ce406004",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Peaks before filtering:  25905\n",
      "Peaks with invalid chr_name:  0\n",
      "Peaks with invalid length:  0\n",
      "Peaks after filtering:  25905\n"
     ]
    }
   ],
   "source": [
    "peaks = ma.check_peak_format(peaks, ref_genome, genomes_dir=genomes_directory)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f981b96c-8cc1-4d18-aca3-ab24ad06e226",
   "metadata": {},
   "source": [
    "You can choose to use either a custom TF binding reference or CellOracle’s default motifs during the motif analysis. If you would like to use our default motifs, you can continue to the next step without loading any additional data. I am using the default!!"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b98c4cd2-a816-41c1-be55-e059b454008c",
   "metadata": {},
   "source": [
    "### 3. Instantiate TFinfo object and search for TF binding motifs\n",
    "The motif analysis module has a custom class, TFinfo. The TFinfo object executes the steps below.\n",
    "\n",
    "1. Converts a peak data into a DNA sequences.\n",
    "2. Scans the DNA sequences searching for TF binding motifs.\n",
    "3. Post-processes the motif scan results.\n",
    "4. Converts data into appropriate format. You can convert data into base-GRN. The base GRN data can be formatted as either a python dictionary or pandas dataframe. This output will be the final base GRN used in the GRN model construction step.\n",
    "\n",
    "If your reference genome file are installed in non-default location, please speficy the location using genomes_dir.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "f1e67ead-5608-4a5c-8f31-ed7540e2e6db",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Instantiate TFinfo object\n",
    "tfi = ma.TFinfo(peak_data_frame=peaks,\n",
    "                ref_genome=ref_genome,\n",
    "                genomes_dir=genomes_directory)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d9226e7-63f5-4176-a908-886139d95400",
   "metadata": {},
   "source": [
    "You can specify the TF binding motif data as follows.\n",
    "\n",
    "tfi.scan(motifs=motifs)\n",
    "\n",
    "If you do not specify the motifs or set motifs to None, the default motifs will be loaded automatically.\n",
    "\n",
    "For mouse and human, “gimme.vertebrate.v5.0.” will be used as the default motifs.\n",
    "\n",
    "For another species, the species-specific TF binding motif data extracted from CisBP ver2.0 will be used.\n",
    "\n",
    "NEXT, scan for TF binding motifs. This will take a while!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "77bbc872-9c84-4dfa-94f3-7b75c47a38ce",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif data entered. Loading default motifs for your species ...\n",
      " Default motif for vertebrate: gimme.vertebrate.v5.0. \n",
      " For more information, please see https://gimmemotifs.readthedocs.io/en/master/overview.html \n",
      "\n",
      "Initiating scanner... \n",
      "\n",
      "Calculating FPR-based threshold. This step may take substantial time when you load a new ref-genome. It will be done quicker on the second time. \n",
      "\n",
      "Motif scan started .. It may take long time.\n",
      "\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b37b9c2963bf4e11962e5aa86eb27e04",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "scanning:   0%|          | 0/24574 [00:00<?, ? sequences/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 12min 41s, sys: 24.5 s, total: 13min 6s\n",
      "Wall time: 14min 10s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# Scan motifs. !!CAUTION!! This step may take several hours if you have many peaks!\n",
    "tfi.scan(fpr=0.02,\n",
    "         motifs=None,  # If you enter None, default motifs will be loaded.\n",
    "         verbose=True)\n",
    "\n",
    "# Save tfinfo object\n",
    "tfi.to_hdf5(file_path=f\"{data_dir}/tfinfo.celloracle.tfinfo\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "1508656e-10cc-4e18-9143-565773047656",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>motif_id</th>\n",
       "      <th>factors_direct</th>\n",
       "      <th>factors_indirect</th>\n",
       "      <th>score</th>\n",
       "      <th>pos</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100015519_100016019</td>\n",
       "      <td>GM.5.0.Mixed.0001</td>\n",
       "      <td></td>\n",
       "      <td>SRF, EGR1</td>\n",
       "      <td>7.904716</td>\n",
       "      <td>253</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_100015519_100016019</td>\n",
       "      <td>GM.5.0.Nuclear_receptor.0002</td>\n",
       "      <td>NR2C2</td>\n",
       "      <td>Nr2c2, NR2C2</td>\n",
       "      <td>9.063828</td>\n",
       "      <td>221</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_100015519_100016019</td>\n",
       "      <td>GM.5.0.Myb_SANT.0001</td>\n",
       "      <td>MYBL2, MYBL1</td>\n",
       "      <td>MYBL2, MYBL1</td>\n",
       "      <td>6.630190</td>\n",
       "      <td>441</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_100015519_100016019</td>\n",
       "      <td>GM.5.0.C2H2_ZF.0003</td>\n",
       "      <td>ZBTB6</td>\n",
       "      <td>Zbtb6, ZBTB6</td>\n",
       "      <td>8.347882</td>\n",
       "      <td>425</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_100015519_100016019</td>\n",
       "      <td>GM.5.0.C2H2_ZF.0006</td>\n",
       "      <td>Sp3, KLF5, SP2, SP3, SP1, KLF4, KLF12</td>\n",
       "      <td>Sp9, Klf5, IRF1, Sp8, Klf6, SP7, SP1, SP6, Sp5...</td>\n",
       "      <td>9.288466</td>\n",
       "      <td>156</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     seqname                      motif_id  \\\n",
       "0  chr10_100015519_100016019             GM.5.0.Mixed.0001   \n",
       "1  chr10_100015519_100016019  GM.5.0.Nuclear_receptor.0002   \n",
       "2  chr10_100015519_100016019          GM.5.0.Myb_SANT.0001   \n",
       "3  chr10_100015519_100016019           GM.5.0.C2H2_ZF.0003   \n",
       "4  chr10_100015519_100016019           GM.5.0.C2H2_ZF.0006   \n",
       "\n",
       "                          factors_direct  \\\n",
       "0                                          \n",
       "1                                  NR2C2   \n",
       "2                           MYBL2, MYBL1   \n",
       "3                                  ZBTB6   \n",
       "4  Sp3, KLF5, SP2, SP3, SP1, KLF4, KLF12   \n",
       "\n",
       "                                    factors_indirect     score  pos  strand  \n",
       "0                                          SRF, EGR1  7.904716  253       1  \n",
       "1                                       Nr2c2, NR2C2  9.063828  221      -1  \n",
       "2                                       MYBL2, MYBL1  6.630190  441      -1  \n",
       "3                                       Zbtb6, ZBTB6  8.347882  425       1  \n",
       "4  Sp9, Klf5, IRF1, Sp8, Klf6, SP7, SP1, SP6, Sp5...  9.288466  156      -1  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tfi.scanned_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a831f7a1-b314-4360-8e62-adcc48e2119c",
   "metadata": {},
   "source": [
    "### 4. Filtering motifs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "e8b9834a-4f57-4655-985d-4e599f20a43a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtering finished: 7445888 -> 1557816\n",
      "1. Converting scanned results into one-hot encoded dataframe.\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b4182dbb128d4008b42956338a8a69c8",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/24574 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2. Converting results into dictionaries.\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "380b06c6a5ff40e1b538e2aaa78959e8",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/16080 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0549096d3319475aad3687427642f38e",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/1090 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Reset filtering\n",
    "tfi.reset_filtering()\n",
    "\n",
    "# Do filtering\n",
    "tfi.filter_motifs_by_score(threshold=10)\n",
    "\n",
    "# Format post-filtering results.\n",
    "tfi.make_TFinfo_dataframe_and_dictionary(verbose=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "79d2887f-a41d-4bd3-972a-5eeba9ab6fde",
   "metadata": {},
   "source": [
    "### 5. Get final base GRN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "a6c74acf-a61a-4e02-b3fa-594970a20943",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>peak_id</th>\n",
       "      <th>gene_short_name</th>\n",
       "      <th>9430076c15rik</th>\n",
       "      <th>Ac002126.6</th>\n",
       "      <th>Ac012531.1</th>\n",
       "      <th>Ac226150.2</th>\n",
       "      <th>Afp</th>\n",
       "      <th>Ahr</th>\n",
       "      <th>Ahrr</th>\n",
       "      <th>Aire</th>\n",
       "      <th>...</th>\n",
       "      <th>Znf784</th>\n",
       "      <th>Znf8</th>\n",
       "      <th>Znf816</th>\n",
       "      <th>Znf85</th>\n",
       "      <th>Zscan10</th>\n",
       "      <th>Zscan16</th>\n",
       "      <th>Zscan22</th>\n",
       "      <th>Zscan26</th>\n",
       "      <th>Zscan31</th>\n",
       "      <th>Zscan4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10_100015519_100016019</td>\n",
       "      <td>Kitl</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10_100487279_100487779</td>\n",
       "      <td>Tmtc3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10_100487943_100488443</td>\n",
       "      <td>Cep290</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10_100487943_100488443</td>\n",
       "      <td>Tmtc3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10_100589013_100589513</td>\n",
       "      <td>4930430F08Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1092 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     peak_id gene_short_name  9430076c15rik  Ac002126.6  \\\n",
       "0  chr10_100015519_100016019            Kitl              0           0   \n",
       "1  chr10_100487279_100487779           Tmtc3              0           0   \n",
       "2  chr10_100487943_100488443          Cep290              0           0   \n",
       "3  chr10_100487943_100488443           Tmtc3              0           0   \n",
       "4  chr10_100589013_100589513   4930430F08Rik              0           0   \n",
       "\n",
       "   Ac012531.1  Ac226150.2  Afp  Ahr  Ahrr  Aire  ...  Znf784  Znf8  Znf816  \\\n",
       "0           0           0    0    0     0     0  ...       0     0       0   \n",
       "1           0           0    0    0     0     0  ...       0     0       0   \n",
       "2           1           0    0    0     0     0  ...       0     0       0   \n",
       "3           1           0    0    0     0     0  ...       0     0       0   \n",
       "4           1           0    0    0     0     0  ...       0     0       0   \n",
       "\n",
       "   Znf85  Zscan10  Zscan16  Zscan22  Zscan26  Zscan31  Zscan4  \n",
       "0      0        0        0        0        0        1       0  \n",
       "1      0        0        0        0        0        0       0  \n",
       "2      0        0        0        0        0        0       0  \n",
       "3      0        0        0        0        0        0       0  \n",
       "4      0        0        0        0        0        0       0  \n",
       "\n",
       "[5 rows x 1092 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = tfi.to_dataframe()\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3491b900-d255-490d-b2ce-5eb2e7c49891",
   "metadata": {},
   "source": [
    "### 6. Save results\n",
    "We will use this information when constructing the GRN models later. Save the results."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "aedb2a0e-4495-4112-bc7b-ff59e0a08efc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Save result as a dataframe\n",
    "df = tfi.to_dataframe()\n",
    "df.to_parquet(f\"{data_dir}/base_GRN_dataframe.parquet\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0351d45e-1e4b-4601-851c-fc23d2631bed",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "celloracle_env",
   "language": "python",
   "name": "celloracle_env"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
